File: Python solutions/remove_duplicates_from_sorted_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    # @param head, a ListNode
    # @return a ListNode

    def deleteDuplicates(self, head):
        if head is None:
            return

        last = head
        now = head.next

        while now:
            if last.val == now.val:
                last.next = now.next
                now = now.next
                continue

            last = last.next
            now = now.next
            # now = now.next

        return head

File: CPP solutions/remove_duplicates_from_sorted_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if (!head) return NULL;
        ListNode *last = head;
        ListNode *now  = head->next;

        while (now)
        {
            if (last->val == now->val)
            {
                now = now->next;
                last->next = now;
                continue;
            }

            last = last->next;
            now = now->next;
        }

        return head;
    }
};